tests/Unit/Controllers/DiagramControllerTest.php

```
@@ -616,6 +616,267 @@ public function testGenerateOkrsWithGeminiFailure(): void
+            static fn(): \StratFlow\Services\GeminiService => $gemini
+        );
+
+        try {
+            $ctrl->generateOkrs();
+            $this->fail('Persistence failure should be rethrown.');
+        } catch (\RuntimeException $e) {
+            $this->assertSame('commit failed', $e->getMessage());
+        }
+
+        $this->assertTrue($releaseCalled);
+    }
```

PHPMD (2.15.0)

[warning] The method testGenerateOkrsRollsBackAndReleasesClaimOnPersistenceFailure() has an NPath complexity of 256. The configured NPath complexity threshold is 200. (undefined)

(NPathComplexity)